src/test/java/persistence/study/ReflectionTest.java

```diff
@@ -0,0 +1,112 @@
+            logger.debug(method.getName());
+        }
+
+        Constructor<?>[] declaredConstructors = carClass.getDeclaredConstructors();
+        for (Constructor<?> constructor : declaredConstructors) {
+            logger.debug(constructor.getName());
+        }
+    }
+
+    @Test
+    @DisplayName("test로 시작하는 메서드 실행")
+    void testMethodRun() throws InvocationTargetException, IllegalAccessException {
+        Class<Car> carClass = Car.class;
+        String testName = "test name";
+        int testPrice = 1000;
+        Method[] declaredMethods = carClass.getDeclaredMethods();
+        Car car = new Car(testName, testPrice);
+        for (Method method : declaredMethods) {
+            if (method.getName().startsWith("test")) {
+                Object result = method.invoke(car);
+                if (method.getName().equals("testGetName")) {
+                    assertThat(result).isEqualTo("test : " + testName);
+                }
+                if (method.getName().equals("testGetPrice")) {
+                    assertThat(result).isEqualTo("test : " + testPrice);
+                }
+            }
+        }
+    }
+
+    @Test
+    @DisplayName("@PrintView 애노테이션 메소드 실행")
+    void testAnnotationMethodRun() throws InvocationTargetException, IllegalAccessException {
+        Class<Car> carClass = Car.class;
+        String testName = "test name";
+        int testPrice = 1000;
+        Method[] declaredMethods = carClass.getDeclaredMethods();
+        Car car = new Car(testName, testPrice);
+        boolean invoked = false;
+        for (Method method : declaredMethods) {
+            if (method.isAnnotationPresent(PrintView.class)) {
+                method.invoke(car);
+                invoked = true;
+            }
+        }
+        assertThat(invoked).isTrue();
+    }
+
+    @Test
+    @DisplayName("private field에 값 할당")
+    void privateFieldAccess() throws NoSuchFieldException, IllegalAccessException {
+        Class<Car> carClass = Car.class;
+        Car car = new Car();
+
+        Field nameField = carClass.getDeclaredField("name");
+        Field priceField = carClass.getDeclaredField("price");
+        nameField.setAccessible(true);
+        nameField.set(car, "testName");
+        priceField.setAccessible(true);
+        priceField.set(car, 1000);
+
+        Object name = nameField.get(car);
+        Object price = priceField.get(car);
+
+        assertThat(name).isEqualTo("testName");
+        assertThat(price).isEqualTo(1000);
+        logger.debug(car.toString());
+    }
+
+    @Test
+    @DisplayName("인자를 가진 생성자의 인스턴스 생성")
+    void constructorWithArgs() throws InvocationTargetException, InstantiationException, IllegalAccessException {
+        Class<Car> carClass = Car.class;
+        Constructor<?>[] declaredConstructors = carClass.getDeclaredConstructors();
+        for(Constructor<?> constructor : declaredConstructors) {
+            if(constructor.getParameterCount() > 0) {
```

Comment:
```suggestion
        for (Constructor<?> constructor : declaredConstructors) {
            if (constructor.getParameterCount() > 0) {
```
`cmd + opt + L`을 통해서 코드 포맷팅할 수 있습니다.